CMakeLists.txt

```diff
@@ -85,6 +85,28 @@ if (OE_SGX)
 
   if (BUILD_ENCLAVES AND WIN32)
     # Setup cross compiler (clangw)
+    find_program(CLANG clang)
+    if (CLANG)
+      # Get the list of clang specific defines and search for __clang_major__
+      execute_process(
+        COMMAND cmd.exe /c " clang -dM -E -x c nul | findstr __clang_major__ "
+        RESULT_VARIABLE HAD_ERROR
+        OUTPUT_VARIABLE CONFIG_OUTPUT
+      )
+      if(NOT HAD_ERROR)
+        # Format the output for a list
+        string(REPLACE " " ";" CONFIG_OUTPUT ${CONFIG_OUTPUT})
+        # Get the major version for clang
+        list(GET CONFIG_OUTPUT 2 MAJOR_VERSION)
+        if (MAJOR_VERSION VERSION_LESS 7)
+          message(FATAL_ERROR "Clang version 7.0 or higher is required")
+        endif ()
+      else ()
+        message(FATAL_ERROR "Could not parse clang major version")
+      endif()
+    else ()
+      message(FATAL_ERROR "Clang is required")
```

Comment:
Could the error handling logic be cleaned up? This error is far, far away from the the source origin. Follow this pattern instead:

```
find_program(CLANG clang)
if (NOT CLANG)
  message(FATAL_ERROR "Clang is required to build ELF enclaves on Windows)
endif ()
# Now we can assume we have clang and don't have to nest inside an `if (CLANG)` or an `else` clause
```

Similar pattern can be applied to the rest, e.g. `if (HAD_ERROR) message(FATAL_ERROR) endif () ...`.

Reply:
Thanks a lot for the feedback, I gave it another go can you PTAL?

Reply:
Looks great, thank you!